File: Java/src/Reverse Bits.java
public class Solution
{
  // you need treat n as an unsigned value
  public int reverseBits(int n)
  {
    int ret = 0;
    for (int count = 0; count < 32; count++)
    {
      ret = ret * 2 + ((n >> count) & 1);
    }
    return ret;
  }
}

File: Python/Reverse Bits.py
class Solution:
    # @param n, an integer
    # @return an integer
    def reverseBits(self, n):
        ret = 0
        for count in range(32):
            if ( n % 2 != 0):
                ret += 2 ** (31-count)
            n = n / 2
        return ret
        